aip/0144.md

```diff
@@ -0,0 +1,104 @@
+---
+aip:
+  id: 144
+  state: reviewing
+  created: 2020-03-19
+permalink: /144
+redirect_from:
+  - /0144
+---
+
+# Repeated fields
+
+Representing lists of data in an API is trickier than it often appears. Users
+often need to modify lists in place, and longer data series within a single
+resource pose a challenge for pagination.
+
+## Guidance
+
+Resources **may** use repeated fields where appropriate.
+
+```proto
+message Book {
+  string name = 1;
+  repeated string authors = 2;
+}
+```
+
+- Repeated fields **must** use a plural field name.
+  - If the English singular and plural words are identical ("moose", "info"),
+    the dictionary word **must** be used rather than attempting to coin a new
+    plural form.
+- Repeated fields **should** have an upper bound that will not cause a single
```

Comment:
I am okay with a judgment call here, which is why I did not specify. Do you believe we need the stricter one?

Reply:
I think it would be worth making that an explicit judgement call of the API producer, rather than leaving it ambiguous in the AIP. Basically, if I find myself asking a question about what the AIP means, I tend to think we should give an answer (even if it's "we don't dictate one way or the other") within the AIP.

Reply:
Fair enough. Since this is **should** and not **must**, I decided to go with the more strict variant (enforced).